nil pointer in Go doesn't generate a panic

#### Steps to reproduce
Run this program in harvey:
```Go
package main

func main() {
	var p *int
	println(*p)
}
```
Built on linux using Go 1.15.2: `GOOS=plan9 go build nil.go`


#### Expected behavior

It should generate a go panic. On 9front:
```
cpu% ./nil
panic: runtime error: invalid memory address or nil pointer dereference
[signal sys: trap: fault read code=0x0 addr=0x0 pc=0x258d44]

goroutine 1 [running]:
main.main()
	/home/fhs/tmp/nil.go:5 +0x24
```

#### Actual behavior

```
cpu% ./nil
nil 173: suicide: sys: trap: fault read addr=0x23 pc=0x2295cf
```

--- a/amd64/include/ureg.h
+++ b/amd64/include/ureg.h
@@ -24,13 +24,13 @@ struct Ureg {
 	uint64_t	r14;
 	uint64_t	r15;
 
-  // these are pointless and I don't want to waste a second on them.
-  // I'm sure it will break something but let's see.
-  // Not pushing them simplifies the interrupt handler a lot.
-  //	uint16_t	ds;
-  //	uint16_t	es;
-  //	uint16_t	fs;
-  //	uint16_t	gs;
+	// These are pointless and not pushing them would simplify the
+	// interrupt handler a lot. However, removing them would break
+	// compatibility with Go Plan 9 binaries.
+	uint16_t	ds;
+	uint16_t	es;
+	uint16_t	fs;
+	uint16_t	gs;
 
 	uint64_t	type;
 	uint64_t	error;				/* error code (or zero) */

--- a/sys/src/9/amd64/l64idt.S
+++ b/sys/src/9/amd64/l64idt.S
@@ -85,6 +85,7 @@ _intrcommon:
 	SWAPGS
 
 _intrnested:
+	SUBQ	$8, %rsp				/* [DEFG]S */
 	PUSHQ	%r15
 	PUSHQ	%r14
 	PUSHQ	%r13
@@ -126,6 +127,7 @@ _intrr:
 	POPQ	%r13
 	POPQ	%r14
 	POPQ	%r15
+	ADDQ	$8, %rsp				/* [DEFG]S */
 
 	CMPW	$SSEL(SiCS, SsTIGDT|SsRPL0), 24(%rsp) 	/* old CS */
 	JE	_iretnested

--- a/sys/src/9/amd64/l64vsyscall.S
+++ b/sys/src/9/amd64/l64vsyscall.S
@@ -50,7 +50,7 @@ syscallentry:
 
 	movq	%gs:32, %r15 	/* restore r15 from m->rathole */
 
-	SUBQ	$(17*8), %rsp
+	SUBQ	$(18*8), %rsp
 
 	MOVQ	%rAX, (0*8)(%rsp)	// ureg.ax
 	MOVQ	%rBX, (1*8)(%rsp)	// ureg.bx
@@ -67,8 +67,12 @@ syscallentry:
 	MOVQ	%r13, (12*8)(%rsp)
 	MOVQ	%r14, (13*8)(%rsp)
 	MOVQ	%r15, (14*8)(%rsp)
-	// (15*8)(%rsp)	// ureg.type
-	// (16*8)(%rsp) // ureg.error
+	// (15*8+0)(%rsp) // ureg.ds
+	// (15*8+2)(%rsp) // ureg.es
+	// (15*8+4)(%rsp) // ureg.fs
+	// (15*8+6)(%rsp) // ureg.gs
+	// (16*8)(%rsp)	// ureg.type
+	// (17*8)(%rsp) // ureg.error
 
 	MOVQ	%rsp, %rsi 				/* Ureg* */
 	// system call number is in %rax, as per linux.
@@ -98,7 +102,7 @@ syscallreturn:
 	MOVQ	(13*8)(%rsp),%r14
 	MOVQ	(14*8)(%rsp),%r15
 
-	ADDQ	$(17*8), %rsp			/* registers + arguments */
+	ADDQ	$(18*8), %rsp			/* registers + arguments */
 
 	CLI
 	SWAPGS

--- a/sys/src/9/amd64/trap.c
+++ b/sys/src/9/amd64/trap.c
@@ -581,6 +581,10 @@ dumpgpr(Ureg *ureg)
 	print("r13\t%#16.16llx\n", ureg->r13);
 	print("r14\t%#16.16llx\n", ureg->r14);
 	print("r15\t%#16.16llx\n", ureg->r15);
+	print("ds\t%#4.4ux\n", ureg->ds);
+	print("es\t%#4.4ux\n", ureg->es);
+	print("fs\t%#4.4ux\n", ureg->fs);
+	print("gs\t%#4.4ux\n", ureg->gs);
 	print("type\t%#llx\n", ureg->type);
 	print("error\t%#llx\n", ureg->error);
 	print("pc\t%#llx\n", ureg->ip);

